src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2443,6 +2443,18 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                     GenerateImplicitConversionError(diagnostics, expression.Syntax, conversion, expression, targetType);
                 }
 
+                if (expression.Kind == BoundKind.TupleLiteral)
+                {
+                    // A BoundTupleLiteral must always be upgraded to a BoundConvertedTupleLiteral during conversion
+                    var sourceTuple = (BoundTupleLiteral)expression;
+                    expression = new BoundConvertedTupleLiteral(
+                        sourceTuple.Syntax,
+                        sourceTuple.Type,
+                        sourceTuple.Arguments,
```

Comment:
>sourceTuple.Arguments, [](start = 24, length = 22)

Would this give us the right information about conversion of the elements? Perhaps we should simply call CreateConversion rather than trying to build BoundConversion node manually.